src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,30 @@
+import { useRef, useEffect } from 'react'
+
+type Timer = ReturnType<typeof setTimeout>
+type SomeFunction = (...args: any[]) => void
+/**
+ *
+ * @param func The original, non debounced function (You can pass any number of args to it)
+ * @param delay The delay (in ms) for the function to return
+ * @returns The debounced function, which will run only if the debounced function has not been called in the last (delay) ms
+ */
+
+export function useDebounce<Func extends SomeFunction>(func: Func, delay = 300) {
+  const timer = useRef<Timer>()
```

Comment:
It would be better to start this with null for better type safety
```suggestion
  const timer = useRef<Timer | null>(null)
```